graphql_compiler/query_formatting/common.py

```diff
@@ -1,22 +1,109 @@
 # Copyright 2017-present Kensho Technologies, LLC.
 """Safely insert runtime arguments into compiled GraphQL queries."""
+import datetime
+import decimal
+
+import arrow
+from graphql import GraphQLBoolean, GraphQLFloat, GraphQLID, GraphQLInt, GraphQLList, GraphQLString
 import six
 
 from ..compiler import GREMLIN_LANGUAGE, MATCH_LANGUAGE, SQL_LANGUAGE
+from ..compiler.helpers import strip_non_null_from_type
 from ..exceptions import GraphQLInvalidArgumentError
+from ..schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
 from .gremlin_formatting import insert_arguments_into_gremlin_query
 from .match_formatting import insert_arguments_into_match_query
 from .sql_formatting import insert_arguments_into_sql_query
 
 
+def _check_is_string_value(value):
+    """Raise if the value is not a proper utf-8 string."""
+    if not isinstance(value, six.string_types):
+        if isinstance(value, bytes):  # should only happen in py3
+            value.decode('utf-8')  # decoding should not raise errors
+        else:
+            raise GraphQLInvalidArgumentError(u'Attempting to convert a non-string into a string: '
+                                              u'{}'.format(value))
+
+
 ######
 # Public API
 ######
 
+
+def validate_argument_type(expected_type, value):
+    """Check if the value is appropriate for the type and usable in any of our backends.
+
+    Backends are the database languages we have the ability to compile to, like OrientDB Match,
+    Gramlin, or SQLAlchemy. This function should be stricter than the validation done by any
+    specific backend. That way code that passes validation can be compiled to any backend.
+
+    Args:
+        expected_type: GraphQL type we expect
+        value: object that can be interpreted as being of that type
+    """
+    if GraphQLString.is_same_type(expected_type):
+        _check_is_string_value(value)
+    elif GraphQLID.is_same_type(expected_type):
+        # IDs can be strings or numbers, but the GraphQL library coerces them to strings.
+        # We will follow suit and treat them as strings.
```

Comment:
It might be time to relax this and allow `int` as well. So the types accepted would be `six.string_types + (int,)`

Reply:
Let's do that when seperately, when we actually implement int parsing of ids?